pkg/controller/servicebindingrequest/service.go

```diff
@@ -170,11 +170,11 @@ func getOwnedResources(
 		if err != nil {
 			return resources, err
 		}
-		for _, item := range lst.Items {
+		for idx, item := range lst.Items {
```

Comment:
The change fixes truly a bug, it does not introduce a new behaviour, more details about it on this [stackoverflow answer](https://stackoverflow.com/a/48826629). As stated there, the proposed solution might have an issue with garbage collection, and perhaps we could solve slightly different:

```diff
diff --git a/pkg/controller/servicebindingrequest/service.go b/pkg/controller/servicebindingrequest/service.go
index 9895a94..6d79132 100644
--- a/pkg/controller/servicebindingrequest/service.go
+++ b/pkg/controller/servicebindingrequest/service.go
@@ -161,10 +161,10 @@ func getOwnedResources(
 	name string,
 	uid types.UID,
 ) (
-	[]*unstructured.Unstructured,
+	[]unstructured.Unstructured,
 	error,
 ) {
-	var resources []*unstructured.Unstructured
+	var resources []unstructured.Unstructured
 	for _, br := range bindableResources {
 		lst, err := client.Resource(br.gvr).Namespace(ns).List(metav1.ListOptions{})
 		if err != nil {
@@ -174,7 +174,7 @@ func getOwnedResources(
 			owners := item.GetOwnerReferences()
 			for _, owner := range owners {
 				if owner.UID == uid {
-					resources = append(resources, &item)
+					resources = append(resources, item)
 				}
 			}
 		}
@@ -202,7 +202,7 @@ func buildOwnedResourceContext(
 
 func buildOwnedResourceContexts(
 	client dynamic.Interface,
-	objs []*unstructured.Unstructured,
+	objs []unstructured.Unstructured,
 	ownerEnvVarPrefix *string,
 	restMapper meta.RESTMapper,
 ) ([]*ServiceContext, error) {
@@ -215,7 +215,7 @@ func buildOwnedResourceContexts(
 			}
 			svcCtx, err := buildOwnedResourceContext(
 				client,
-				obj,
+				&obj,
 				ownerEnvVarPrefix,
 				restMapper,
 				br.inputPath,


```

Reply:
as a followup, we should inspect the code in order to see if we have the same issue elsewhere.

Reply:
raised jira ticket https://issues.redhat.com/browse/APPSVC-611